crates/inspection-service/src/inspection_service.rs

```diff
@@ -85,57 +93,105 @@ async fn serve_requests(
     req: Request<Body>,
     node_config: NodeConfig,
 ) -> Result<Response<Body>, hyper::Error> {
-    let mut resp = Response::new(Body::empty());
-    match (req.method(), req.uri().path()) {
-        // Expose the node configuration
-        (&Method::GET, "/configuration") => {
+    // Process the request and get the response components
+    let (status_code, body, content_type) = match req.uri().path() {
+        "/configuration" => {
+            // Exposes the node configuration
             if node_config.inspection_service.expose_configuration {
                 // We format the configuration using debug formatting. This is important to
                 // prevent secret/private keys from being serialized and leaked (i.e.,
                 // all secret keys are marked with SilentDisplay and SilentDebug).
                 let encoded_configuration = format!("{:?}", node_config);
-                *resp.body_mut() = Body::from(encoded_configuration);
+                (
+                    StatusCode::OK,
```

Comment:
why directly return a tuple instead of modifing resp to take statusCode and content_type_text?

Reply:
I originally did that, but it results in slightly more messy code, e.g., (i) you can't modify the content type header of a response with a single call, instead you need to call `headers_mut()` and overwrite all the headers in one instance (which is a bit ugly); and (ii) you end up with more code repetition in each branch because of the overwrites. So, I decided to take the quick route.

I'm sure there's some way of cleaning it up, but I also don't want to waste too much time on this until we actually revisit the service and likely re-write most of it anyway 😄